urlcdab7d8364_jrwalsh_CycTools_tgz-pJ8-edu_iastate_cyctools_MainJ8/src/edu/iastate/cyctools/tools/load/model/BatchUpdate.java

```diff
@@ -0,0 +1,289 @@
+	private ArrayList<Event> eventLog;
+    
+    public BatchUpdate(ArrayList<AbstractFrameEdit> frameEditList, LoadPanel loadPanel) {
+    	this.loadPanel = loadPanel;
+    	initDefault();
+    	this.frameEdits = frameEditList;
+    	
+    	frameEditsMap = new HashMap<String, ArrayList<AbstractFrameEdit>>();
+    	frameIDSet = new TreeSet<String>();
+		for (AbstractFrameEdit frameEdit : frameEdits) {
+			for (int row : frameEdit.getAssociatedRows()) {
+				lines.add(new Integer(row));
+			}
+			
+			if (frameEditsMap.containsKey(frameEdit.getFrameID())) {
+				ArrayList<AbstractFrameEdit> frameEditArray = frameEditsMap.get(frameEdit.getFrameID());
+				frameEditArray.add(frameEdit);
+				frameEditsMap.put(frameEdit.getFrameID(), frameEditArray);
+			} else {
+				ArrayList<AbstractFrameEdit> frameEditArray = new ArrayList<AbstractFrameEdit>();
+				frameEditArray.add(frameEdit);
+				frameEditsMap.put(frameEdit.getFrameID(), frameEditArray);
+			}
+			frameIDSet.add(frameEdit.getFrameID());
+		}
+    }
+    
+    public void initDefault() {
+    	linesProcessed = new TreeSet<Integer>();
+    	lines = new TreeSet<Integer>();
+    	eventLog = new ArrayList<Event>();
+    }
+    
+    // Accessors
+    public ArrayList<AbstractFrameEdit> getFrameEdits() {
+    	return frameEdits;
+    }
+    
+    public HashMap<String, ArrayList<AbstractFrameEdit>> getFrameEditsMap() {
+    	return frameEditsMap;
+    }
+    
+    public TreeSet<String> getFrameIDSet() {
+    	return frameIDSet;
+    }
+    
+    public int getLines() {
+    	return lines.size();
+    }
+    
+    public ArrayList<Event> getEventLog() {
+    	return eventLog;
+    }
+    
+    public DefaultListModel<String> getFrameIDsModel() {
+    	DefaultListModel<String> listModel = new DefaultListModel<String>();
+		for (String frameID : getFrameIDSet()) {
+			listModel.addElement(frameID);
+		}
+		return listModel;
+    }
+    
+    public void downloadFrames(JavacycConnection conn) {
+    	progressMonitor = new ProgressMonitor(DefaultController.mainJFrame, "Analyzing Imports...", "", 0, 100);
+    	task = new DownloadFramesTask(conn, progressMonitor, frameIDSet);
+		
+		task.addPropertyChangeListener(new PropertyChangeListener() {
+            @Override
+            public void propertyChange(PropertyChangeEvent e) {
+                if ("progress".equals(e.getPropertyName())) {
+                	progressMonitor.setProgress((Integer) e.getNewValue());
+                	
+                	if (progressMonitor.isCanceled()) {
+                		task.cancel(true);
+                	}
+                	
+                	if (task.isDone() && !task.isCancelled()) {
+                		try {
+							framesFromKB = task.get();
+							progressMonitor.setProgress(0);
+							progressMonitor.close();
+							loadPanel.showPreviewPanel();
+							loadPanel.addAuthorCredits();
+							DefaultController.mainJFrame.getGlassPane().setVisible(false);
+						} catch (InterruptedException e1) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)